src/ideal/ideal.jl

```diff
@@ -72,6 +72,21 @@ iszero(I::sideal) = Bool(libSingular.idIs0(I.ptr))
 """
 iszerodim(I::sideal) = Bool(libSingular.id_IsZeroDim(I.ptr, base_ring(I).ptr))
 
+@doc Markdown.doc"""
+    dimension(I::sideal{T})
+> Given an ideal $I$ this function computes the Krull dimension
+> of the ring $R/I$, where $R$ is the polynomial ring over
+> which $I$ is an ideal. The ideal must be over a polynomial ring
+> over a field, and a Groebner basis.
+"""
+function dimension(I::sideal{T}) where T <: Singular.RingElem
```

Comment:
why not have one version with T <: Singular.FieldElem which is documented as such and another one for Singular.RingElement which only raises an error (if you really require this)